.github/workflows/linux.yml

```diff
@@ -34,23 +40,54 @@ jobs:
         sudo apt-get update
         sudo apt-get install -y libopenblas-dev libatlas-base-dev liblapack-dev gfortran libgmp-dev libmpfr-dev libsuitesparse-dev ccache libmpc-dev
 
+    - name: Caching Python dependencies
+      uses: actions/cache@v2
+      id: cache
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip
+
     - name: Install Python packages
       run: |
         python -m pip install numpy setuptools wheel cython pytest pytest-xdist pybind11 pytest-xdist mpmath gmpy2 pythran ninja
         python -m pip install git+https://github.com/rgommers/meson.git@fix-cython-mixed-sources
 
+    - name:  Prepare compiler cache
+      id:    prep-ccache
+      shell: bash
+      run: |
+        mkdir -p "${CCACHE_DIR}"
+        echo "::set-output name=dir::$CCACHE_DIR"
+        echo "::set-output name=today::$(date -I)"
```

Comment:
Question about this: the way `outputs.today` is used, it seems like your intent here is to make the cache expire after a day? Or maybe I'm not clear on what `restore-keys: cache-linux-` does. 

I would expect this cache to be date/time-independent. It should key on the exact job, which can be done in several different ways, but not be time-dependent.

Reply:
my understanding was that the stem cache-linux will always provide a hit on the MR branch, then the master branch. The timestamp is an attempt the make the most recent cache the preferred one.

Reply:
I talked to @HarshCasper a bit about describing how this works in this PR in more detail. It's very well possible this is a good setup, and I'm just failing to understand the GHA syntax/behavior here.

What I'd like to understand is how this works with many PRs in parallel as well as merges to master. Ideally the cache should be shared as well as possible between different PRs, merges to master, and also work when there's >1 day between consecutive pushes to the same PR (you can assume the CI system merges the most recent master commit into the PR branch).